pkg/restore/systable_restore.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 PingCAP, Inc. Licensed under Apache-2.0.
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/parser/mysql"
+	filter "github.com/pingcap/tidb-tools/pkg/table-filter"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+
+	berrors "github.com/pingcap/br/pkg/errors"
+	"github.com/pingcap/br/pkg/logutil"
+	"github.com/pingcap/br/pkg/utils"
+)
+
+// RestoreSystemSchemas restores the system schema(i.e. the `mysql` schema).
+// Detail see https://github.com/pingcap/br/issues/679#issuecomment-762592254.
+func (rc *Client) RestoreSystemSchemas(ctx context.Context, f filter.Filter) {
+	sysDB := mysql.SystemDB
+
+	temporaryDB := utils.TemporaryDBName(sysDB)
+	defer rc.cleanTemporaryDatabase(ctx, sysDB)
+
+	if !f.MatchSchema(temporaryDB.O) {
+		log.Debug("system database filtered out", zap.String("database", sysDB))
+		return
+	}
+	originDatabase, ok := rc.databases[temporaryDB.O]
+	if !ok {
+		log.Info("system database not backed up, skipping", zap.String("database", sysDB))
+		return
+	}
+	db, ok := rc.getDatabaseByName(sysDB)
+	if !ok {
+		// Or should we create the database here?
+		log.Warn("target database not exist, aborting", zap.String("database", sysDB))
+		return
+	}
+
+	tablesRestored := make([]string, 0, len(originDatabase.Tables))
+	for _, table := range originDatabase.Tables {
+		tableName := table.Info.Name
+		if f.MatchTable(sysDB, tableName.O) {
+			rc.replaceTemporaryTableToSystable(ctx, tableName.L, db)
+		}
+		tablesRestored = append(tablesRestored, tableName.L)
+	}
+	if err := rc.afterSystemTablesReplaced(ctx, tablesRestored); err != nil {
+		for _, e := range multierr.Errors(err) {
+			logutil.WarnTerm("error during winding up the restoration of system table", zap.String("database", sysDB), logutil.ShortError(e))
```

Comment:
what is meant by "wind up the restoration" 🤔

Reply:
> From New Oxford American Dictionary, `wind up'.
> gradually or finally bring an activity to a conclusion: *the experiments had to be wound up because the funding stopped.*

I was trying to find a word to describe the works for making the system config active after replacing tuples into system tables, but seems this verb is too exotic in some way... 🥲 Maybe just "activating new config" could be fine.

Reply:
or maybe just "error reconfigurating the system tables"